### PR TITLE
support more request params in client

### PR DESCRIPTION
**Pull request by constb**
Hello, me again. :) I needed to add extra parameters to http requests in RPC client:

* http basic auth
* setting `X-Api-Key` header on request
* using custom ssl certificated in https agent

Instead of making many little patches, I made this one to expose all available http.request() options at once.

I also used this opportunity to make some little fixes: typo in utils method name, merging all chunks at once at the end, so less memory allocations and also `options` object is created once for all requests. I also made sure to keep backwards compatibility with code that uses `timeout` option.

### CODE REVIEW

File: src/client/index.js [hunk 1]
```diff
@@ -1,59 +1,54 @@
...
-    timeout: 0
+    path: '/',
+    connectTimeout: 0,
+    responseTimeout: 0
 };
 
 class Client {
     constructor(userOptions) {
         this.options = Object.assign({}, defaultOptions, userOptions || {});
+        this.options.headers = Object.assign({}, this.options.headers || {}, { 'Content-Type': 'application/json' });
```
Comment (lindell) [inline src/client/index.js]: I see no reason to disallow a custom `Content-Type`. Even if I see no reason to change it, I think it should be allowed 🙂

File: README.md [hunk 2]
```diff
@@ -60,8 +60,11 @@ try {
...
-| `host`    | `"127.0.0.1"` | The host that the server listen on |
-| `port`    | `6356`        | The port that the server listen on |
-| `timeout` | `0`           | The request timeout                |
\ No newline at end of file
+| Option            | Default       | Description                         |
+| ----------------- | ------------- | ----------------------------------- |
+| `host`            | `"127.0.0.1"` | The host that the server listens on |
+| `port`            | `6356`        | The port that the server listens on |
+| `connectTimeout`  | `0`           | The socket connection timeout       |
+| `responseTimeout` | `0`           | The response wait timeout           |
```
Comment (lindell) [inline README.md]: The header option should be present in the options table

**Comment by constb** [comment]
Done. :)

File: README.md [hunk 3]
```diff
@@ -60,11 +60,12 @@ try {
...
-| `port`            | `6356`        | The port that the server listens on |
-| `connectTimeout`  | `0`           | The socket connection timeout       |
-| `responseTimeout` | `0`           | The response wait timeout           |
-
-Client's constructor also supports options from [http.request()](https://nodejs.org/api/http.html#http_http_request_options_callback). For example, you can set `headers` to add extra headers to http request, or `auth` if you need Basic Authentication. You cannot change _http method_ or `Content-Type` header.
+| Option            | Default                     | Description                         |
+| ----------------- | --------------------------- | ----------------------------------- |
+| `host`            | `"127.0.0.1"`               | The host that the server listens on |
+| `port`            | `6356`                      | The port that the server listens on |
+| `headers`         | `{ 'X-Api-Key': 'secret' }` | Additional request headers          |
```
Comment (lindell) [inline README.md]:
```js
{ 'Content-Type': 'application/json' }
```
Is the default

Comment (lindell) [reply README.md]: Or potentially `{}`

**Comment by lindell** [comment]
Merged. Thanks for the PR :)

**Pull request merged**